internal/monitor/trino.go

```diff
@@ -0,0 +1,224 @@
+package monitor
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/razorpay/trino-gateway/internal/boot"
+	"github.com/razorpay/trino-gateway/internal/provider"
+	"github.com/trinodb/trino-go-client/trino"
+)
+
+type ITrinoClient interface {
+	IsClusterUp(ctx *context.Context) (bool, error)
+	RunQuery(ctx *context.Context, query string) (*sql.Rows, error)
+}
+
+type TrinoClient struct {
+	db   *sql.DB
+	url  url.URL
+	user string
+	pass string
+}
+
+func NewTrinoClient(ctx *context.Context, url url.URL, user string) *TrinoClient {
+	return &TrinoClient{
+		url:  url,
+		user: user,
+	}
+}
+
+func (t *TrinoClient) httpClient(ctx *context.Context) *http.Client {
+	return &http.Client{
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			DialContext: (&net.Dialer{
+				Timeout:   5 * time.Second,
+				KeepAlive: 5 * time.Second,
+			}).DialContext,
+			MaxIdleConns:          3,
+			IdleConnTimeout:       10 * time.Second,
+			TLSHandshakeTimeout:   5 * time.Second,
+			ExpectContinueTimeout: 2 * time.Second,
+		},
+	}
+}
+
+func (t *TrinoClient) initDbClient(ctx *context.Context) error {
```

Comment:
nit: Suggesting to rename this to `initTrinoDbClient` or `initTrinoClient` or just `init`

Reply:
renamed to initTrinoDbClient